components/messenger.rst

```diff
@@ -38,11 +38,28 @@ Concepts
    something can be a message broker or a third party API for example.
 
 **Receiver**:
-   Responsible for deserializing and forwarding messages to handler(s). This
-   can be a message queue puller or an API endpoint for example.
+   Responsible for retrieving, deserializing and forwarding messages to handler(s).
+   This can be a message queue puller or an API endpoint for example.
 
 **Handler**:
    Responsible for handling messages using the business logic applicable to the messages.
+   Handlers are called by the ``HandleMessageMiddleware`` middleware.
+
+**Middleware**:
+   Man-in-the-middle classes, allowing to spy the message and its wrapper (the envelope).
```

Comment:
This phrase sounded a bit weird to me, so I asked on Symfony's Slack and our English native friend Ciaran suggested to reword it as:

```
Intermediary classes, which can inspect the message and its wrapper (the envelope).
```

Reply:
Thanks! But is `intermediary` really meaningful regarding the concept?

I also found online something we may adapt:

> Middleware (literally the software in the middle) are not core concerns (business logic) of an application. Instead, they are cross cutting concerns applicable throughout the application and affecting the entire message bus.

Reply:
A bit verbose, but what about:

```rst
   Middleware can access the message and its wrapper (the envelope) while it is
   dispatched through the bus.
   Literally *"the software in the middle"*, those are not about core concerns
   (business logic) of an application. Instead, they are cross cutting concerns
   applicable throughout the application and affecting the entire message bus.
   For instance: logging, validating a message, starting a transaction, ...
   They are also responsible for calling the next middleware in the chain,
   which means they can tweak the envelope, by adding items to it, or even
   replacing it and the message inside, as well as interrupt the middleware chain.
```

?